src/v/cluster/service.cc

```diff
@@ -461,10 +469,16 @@ service::do_collect_node_health_report(get_node_health_request req) {
           .error = map_health_monitor_error_code(res.error())};
     }
     auto report = std::move(res.value());
-    // clear all revision ids to prevent sending them to old redpanda nodes
-    if (req.decoded_version == 0) {
+    // clear all revision ids to prevent sending them to old versioned redpanda
+    // nodes
+    if (req.decoded_version == get_node_health_request::initial_version) {
```

Comment:
nit: can use `req.decoded_version > get_node_health_request::revision_id_version` here for symmetry

Reply:
Done